src/wasm/wasm-binary.cpp

```diff
@@ -2505,22 +2531,35 @@ void WasmBinaryBuilder::readTableElements() {
   }
   for (size_t i = 0; i < numSegments; i++) {
     auto headerFormat = getU32LEB();
-    if (headerFormat != 0) {
+    if (headerFormat != 0x0 && headerFormat != 0x2) {
```

Comment:
These constants should probably be given names in `wasm-binary.h`.

Reply:
So there are 8 kinds of header formats defined in the reference types proposal, which are a bit hard to come up with good names for; e.g. `ActiveSegmentWithRefInstructionWithoutTableIndex` would be one. The header byte is basically a bitfield, so we're probably going to parse it that way and not write separate code for each of the 8 different kinds.

Here I wrote it like this because I just wanted a minimal implementation to be able to test multi-table setups.

Reply:
It would be reasonable to have an enum giving names to each bit, then. Something like this (with better names, of course):

```
enum TableHeader {
  BitZeroMeaning = 1,
  BitOneMeaning = 1 << 1,
  ...
};
```

Reply:
The second bit changes meaning based on the first bit (!), but sure this would be better I guess. From the [reference types spec](https://webassembly.github.io/reference-types/core/binary/modules.html#element-section):

> The initial byte can be interpreted as a bitfield. Bit 0 indicates a passive or declarative segment, bit 1 indicates the presence of an explicit table index for an active segment and otherwise distinguishes passive from declarative segments, bit 2 indicates the use of element type and element expressions instead of element kind and element indices.

Reply:
Yuck, maybe a hybrid approach would work then? We could have 4 enum values covering all combinations of the first 2 bits, and a single enum value for bit 2,

Reply:
Take a look at my last commit I pushed just now. Doesn't look bad, but let me know what you think.